src/core/util.c

```diff
@@ -26,16 +26,22 @@ unsigned long long Pagesize;
 unsigned long long Mmap_align;
 
 #if ANY_VG_TOOL_ENABLED
-/* initialized to true if the process is running inside Valgrind */
+/*
+ * Initialized to true if the process is running inside Valgrind, EXCLUDING
```

Comment:
I don't know, I don't like this that much. Can we make this into a flag so that you can distinguish pmemcheck with `On_valgrind & VALGRIND_PMEMCHECK` ?

Reply:
Coding it like that will generate more instructions and detecting !pmemcheck will be quite complex (On_valgrind && ~(On_valgrind & VALGRIND_PMEMCHECK)).

Reply:
What would generate more instructions? All existing code paths will stay the same (because they just check that a variable is != 0). Only in the code paths that we want to detect the presence of a particular tool, this would require literally one `&` operator.
But ok. What's the reason for setting on_valgrind to 0 for pmemcheck? I understand that ideally, we'd only issue the required assembly under required tools - but this patch does this halfway anyway. Following this logic, we should also add `On_memcheck`, `On_drd`, and `On_helgrind` and then remove `On_valgrind` altogether.